automarker_2/automarker_app/lib/steps.py

```diff
@@ -673,11 +685,16 @@ class PythonRunPytests(Step):
     name = "run learner pytests"
 
     def run(self, project_uri, clone_dir_path, self_test, config, fail_fast):
+        python_executable_path_tests = (
+            f"{clone_dir_path}/{PYTHON_EXECUTABLE_PATH}"
+            if is_current_os_windows()
+            else PYTHON_EXECUTABLE_PATH
+        )
+
         command = (
-            f"cd {clone_dir_path} && automarker_venv/bin/python -m pytest --tb=line"
+            f"cd {clone_dir_path} && {python_executable_path_tests} -m pytest --tb=line"
```

Comment:
Something is off here, original logic was like this:

```python
command = f"cd {clone_dir_path} && automarker_venv/bin/python -m pytest --tb=line"
```

Meaning we'd first `cd` into the cloned repo, then run pytest using the python executable. So to do the same we can just replace the "bin/python" part as this the executable does not sit directly in the `clone_dir_path` as we've assumed here.

```python
command = f"cd {clone_dir_path} && automarker_venv/{PYTHON_EXECUTABLE_PATH} -m pytest --tb=line"
```

Reply:
I just realized you are right, `&&` won't work on Windows. So, I think to clean up our code, we can cater to it this way:

```python
command = f"{clone_dir_path/'automarker_venv'/PYTHON_EXECUTABLE_PATH} -m pytest {clone_dir_path/tests --tb=line"
```

I'm just trying not to repeat some stuff, so I'm thinking we can curb that by passing the tests directory to the same exec. path we use in other places. Let me know what you think.

Unfortunately I've got no Windows box I can use to test this right away